src/test/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/PerformerNameAndAddressTest.java

```diff
@@ -0,0 +1,99 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+import org.junit.jupiter.api.Test;
+
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.EdifactValidationException;
+
+class PerformerNameAndAddressTest {
+
+    private final PerformerNameAndAddress performingOrganisationNameAndAddress = PerformerNameAndAddress.builder()
+        .performingOrganisationName("LONDON CITY HOSPITAL")
+        .build();
+
+    private final PerformerNameAndAddress performerNameAndAddress = PerformerNameAndAddress.builder()
+            .identifier("A2442389")
+            .code(HealthcareRegistrationIdentificationCode.CONSULTANT)
+            .performerName("DR J SMITH")
+            .build();
+
+    @Test
+    void when_edifactStringDoesNotStartWithCorrectKey_expect_illegalArgumentExceptionIsThrown() {
+        assertThatThrownBy(() -> PerformerNameAndAddress.fromString("wrong value"))
+            .isExactlyInstanceOf(IllegalArgumentException.class)
+            .hasMessage("Can't create PerformerNameAndAddress from wrong value");
+    }
+
+    @Test
+    void when_edifactStringIsPassed_expect_returnAPerformingOrganisationNameAndAddressObject() {
+        assertAll(
+            () -> assertThat(performingOrganisationNameAndAddress)
+                    .usingRecursiveComparison()
+                    .isEqualTo(PerformerNameAndAddress.fromString("NAD+SLA+++LONDON CITY HOSPITAL")),
+
+            () -> assertThat(performerNameAndAddress)
+                    .usingRecursiveComparison()
+                    .isEqualTo(PerformerNameAndAddress.fromString("NAD+SLA+A2442389:902++DR J SMITH"))
+        );
+    }
+
+    @Test
+    void when_buildingSegmentObjectWithEmptyHealthcareCode_expect_illegalArgumentExceptionIsThrown() {
+        assertThatThrownBy(
+            () -> PerformerNameAndAddress.builder()
+                .identifier("A2442389")
+                .code(HealthcareRegistrationIdentificationCode.fromCode(""))
+                .performingOrganisationName("")
+                .performerName("")
+                .build()
+        )
+            .isExactlyInstanceOf(IllegalArgumentException.class)
+            .hasMessage("No HealthcareRegistrationIdentificationCode for ''");
```

Comment:
How much of this is actually needed to trigger the expected behaviour? Could we strip anything out? If all of those fields are necessary, the condition might be better communicated by preparing the irrelevant stuff outside of the assertion:
```suggestion
        final var builder = PerformerNameAndAddress.builder()
            .identifier("A2442389")
            .performingOrganisationName("")
            .performerName("");

        assertThatThrownBy(() -> builder.code(HealthcareRegistrationIdentificationCode.fromCode("")).build())
            .isExactlyInstanceOf(IllegalArgumentException.class)
            .hasMessage("No HealthcareRegistrationIdentificationCode for ''");
```

Reply:
Done in 4bbb811